src/producer/sendMessages.js

```diff
@@ -67,18 +76,36 @@ module.exports = ({ logger, cluster, partitioner }) => {
         const entries = Array.from(topicMetadata.entries())
         const topicDataForBroker = entries
           .filter(([_, { partitionsPerLeader }]) => !!partitionsPerLeader[broker.nodeId])
-          .map(([topic, { partitionsPerLeader, messagesPerPartition }]) => ({
+          .map(([topic, { partitionsPerLeader, messagesPerPartition, sequencePerPartition }]) => ({
             topic,
             partitions: partitionsPerLeader[broker.nodeId],
+            sequencePerPartition,
             messagesPerPartition,
           }))
 
         const topicData = createTopicData(topicDataForBroker)
 
         try {
-          const response = await broker.produce({ acks, timeout, compression, topicData })
+          const response = await broker.produce({
+            producerId: transactionManager.getProducerId(),
+            producerEpoch: transactionManager.getProducerEpoch(),
+            acks,
+            timeout,
+            compression,
+            topicData,
+          })
+
           const expectResponse = acks !== 0
-          responsePerBroker.set(broker, expectResponse ? responseSerializer(response) : [])
+          const formattedResponse = expectResponse ? responseSerializer(response) : []
+
+          formattedResponse.forEach(({ topicName, partition }) => {
+            const size = topicMetadata.get(topicName).messagesPerPartition[partition].length
+            const previous = topicMetadata.get(topicName).sequencePerPartition[partition]
+
+            transactionManager.updateSequence(topicName, partition, previous + size)
```

Comment:
The sequence number is an `int32`, it will eventually reach the max number. I think we need to rotate the number, similar to what I do with the correlation id.

https://github.com/tulios/kafkajs/blob/095b6e0c5b7623634eb37eca176433808f9d6154/src/network/connection.js#L316-L322

WDYT?

Reply:
Sounds good to me! Worth noting that the Java implementation appears not to bother with this check (will implement nonetheless): https://github.com/apache/kafka/blob/9a0ea25fee85837748145d37c69cf4d9bb7f9933/clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java#L417

Reply:
@tulios Also, `Number.MAX_VALUE` is significantly larger than the maximum (signed) Int32 value (2 ^ 32 vs 2 ^ 1024). AFAIK JS has no predefined constant for Int32, so I'll need to define my own.